### PR TITLE
Removed `ios_static_framework` and `tvos_static_framework` macros.

### PR DESCRIPTION
**Pull request by keith**
This change refactors current ios/tvos static framework macros that are
currently wrapping dependencies under a `apple_static_library` target to
link static libraries, to use `apple_common.link_multi_arch_static_library`
API instead.

Additionally, `deps` cfg was set to use `apple_common.multi_arch_split`,
and the `static_framework_transition` was deprecated, and setting the
`emit_swiftinterface` setting is now done at rule level using the
`apple_rule_transition` using a rule private attribute.

Finally, this change deprecates `swift_static_framework` aspect to use
the `swift_framework` partial instead. Since this partial requires a hash
of SwiftInfo providers keyed by architecture, the `_cpu_string` method
from `transition_support` was exposed to transform link output target
triplets to `apple_common.multi_arch_split` transition keys.

PiperOrigin-RevId: 440942096
(cherry picked from commit 624efd574c0466263158a81efe23f02b03233004)

### CODE REVIEW

**Comment by keith** [comment]
this was rolled back on upstream here 26cf68827c063528161ad77674e916d10414cbfd so i think it'll need some fixes later. either way it's blocked on bazel releases probably

**Comment by keith** [comment]
I was wrong about the rollback order, everything was done again

**Comment by keith** [comment]
this uses the new api that requires bazel 6.x+

File: apple/internal/ios_rules.bzl [hunk 1]
```diff
@@ -1368,25 +1376,29 @@ def _ios_dynamic_framework_impl(ctx):
...
     bundle_name, bundle_extension = bundling_support.bundle_full_name_from_rule_ctx(ctx)
     executable_name = bundling_support.executable_name(ctx)
     features = features_support.compute_enabled_features(
         requested_features = ctx.features,
         unsupported_features = ctx.disabled_features,
     )
-    label = ctx.label
     platform_prerequisites = platform_support.platform_prerequisites_from_rule_ctx(ctx)
-    predeclared_outputs = ctx.outputs
     resource_deps = ctx.attr.deps + ctx.attr.resources
```
[the remark this answers is not in the document]
Comment (keith) [reply apple/internal/ios_rules.bzl:1396]: :ups

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**